oidc.go

```diff
@@ -93,18 +94,23 @@ func NewProvider(ctx context.Context, issuer string) (*Provider, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to read response body: %v", err)
```

Comment:
@pquerna the caller of this function will wrap this error, and will be the one adding the `oidc:` prefix. I'm not usre it's this functions' job to handle putting the source URL into the error, I believe caller's should do that if they're going to.